### DISCUSSION TITLE
Creating new records on Select component doesn't work

### DISCUSSION THREAD

**Discussion by nillegal** [filamentphp/filament#4978] [Help] [answered]
Hello, I'm trying to build a form using the Filament Forms.
I have a `Select` component to allow users to add new values if the value entered doesn't match to any values in the options.

here are the relationships I have set
![code](https://user-images.githubusercontent.com/26672365/202455015-30f6e78f-4db2-4c9d-ac97-6edc56d73d5a.png)

and here's how I try to render the `Select` component
![code 2](https://user-images.githubusercontent.com/26672365/202455125-b9357a1b-b03e-47a8-bf69-62d86840afdd.png)

it actually returns `Call to a member function brand() on null`

Is there something I did wrong?

Filament Forms v2.16
Livewire v2.10
PHP v 8.1.12

**Comment by danharrin** [comment, marked as answer]
Have you defined the `getFormModel()` on the Livewire component? Read the docs about relationships

**Reply by nillegal** [reply]
I missed it @danharrin . Thank you so much

**Reply by nillegal** [reply]
By the way the modal actually doesn't show up. The '+' button just loads but nothing happens, no error returned.
But the `Select` actually works, I can search `brands` through the `brand` relationship

**Reply by danharrin** [reply]
Have you added `{{ $this->modal }}` like the docs say?

**Reply by nillegal** [reply]
Haven't found it in the doc, could you provide the link? 

I did not mention, I'm not using the Admin Panel on this case, just the Form Builder besides Laravel Breeze.

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/forms/fields#creating-new-records

**Reply by nillegal** [reply]
> Have you added `{{ $this->modal }}` like the docs say?

Adding `{{ $this->modal }}` after `{{ $this->form }}` works! I didn't get where to place it. I thought it will automatically be rendered as it is a part of the `Select` component.
